### PR TITLE
boards: add cc1312-launchpad board.

### PR DESCRIPTION
**Pull request by jeandudey**
### Contribution description

It is essentially the same buttons and leds pins, what changes is the
CCXML files used to flash with `make flash`.

### Testing procedure

Go to any examples and execute `make flash BOARD=cc1312-launchpad`, the code should build correctly and get flashed on a connected CC1312 at the port `ttyACM0`.

### Issues/PRs references

See also #9813, #6715.

### CODE REVIEW

**Comment by jeandudey** [comment]
### Question

Actually the CC1312 has available two UARTs, the default UART0 on DIO2 (RXD) and DIO3 (TXD), the other can be chosen arbitrarily (the IOC takes care of multiplexing it to the UART peripheral). Should UART1 be defined with a fixed set of pins or there's a way of changing that on the fly?

**Comment by benpicco** [comment]
Changes look good.
You can update the `Makefile.ci` lists automatically by running

```bash
dist/tools/insufficient_memory/add_insufficient_memory_board.sh cc1312-launchpad
```

**Comment by jeandudey** [comment]
@benpicco Done, thanks for pointing it out :+1: .

**Comment by jeandudey** [comment]
@benpicco For sure, I'm working right now on implementing a radio driver for the cc1312 MCUs, should be possible to adapt to others of the same family, when I get this working I'll make a PR for sure!

File: boards/cc1312-launchpad/Makefile.include [hunk 1]
```diff
@@ -0,0 +1,11 @@
+export XDEBUGGER = XDS110
```
Comment (aabadie) [inline boards/cc1312-launchpad/Makefile.include]: This export is not needed.

File: tests/bench_xtimer/Makefile.ci [hunk 2]
```diff
@@ -0,0 +1,3 @@
+BOARD_INSUFFICIENT_MEMORY := \
```
Comment (aabadie) [inline tests/bench_xtimer/Makefile.ci]: You don't need to add this file for this test. Just add the board in the `LOW_MEMORY_BOARDS` variable.

https://github.com/RIOT-OS/RIOT/blob/0703449c0f58c082ed9dbc0253b89cc6e7ba2992/tests/bench_xtimer/Makefile#L7

File: boards/cc1312-launchpad/Makefile.features [hunk 3]
```diff
@@ -0,0 +1,7 @@
+CPU = cc13x2
+CPU_MODEL = cc1312r1f3
+
+# Put defined MCU peripherals here (in alphabetical order)
+FEATURES_PROVIDED += periph_gpio
```
Comment (aabadie) [inline boards/cc1312-launchpad/Makefile.features]: No `periph_gpio_irq` ?
Support is provided by the CPU:
https://github.com/RIOT-OS/RIOT/blob/0703449c0f58c082ed9dbc0253b89cc6e7ba2992/cpu/cc26xx_cc13xx/periph/gpio.c#L90-L92

File: boards/cc1312-launchpad/include/periph_conf.h [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+*
+* The used CC1312 CPU only supports a single UART device, so all we need to
+* configure are the RX and TX pins.
+*
+* Optionally we can enable hardware flow control, by setting UART_HW_FLOW_CTRL
+* to 1 and defining pins for UART_CTS_PIN and UART_RTS_PIN.
+* @{
+*/
+/**
+ * @name    UART configuration
```
Comment (aabadie) [inline boards/cc1312-launchpad/include/periph_conf.h]: I wondering how this renders in the documentation. I think the 2 `@name  UART configuration` blocks should be merged.

Comment (jeandudey) [reply boards/cc1312-launchpad/include/periph_conf.h]: You're right, they should be merged. [See this](https://doc.riot-os.org/boards_2cc1352-launchpad_2include_2periph__conf_8h.html), it's how that block is rendered for the `cc1352-launchpad` board, it needs to be fixed too. Maybe for another PR, alongside the `export` that is too on that board.

**Comment by jeandudey** [comment]
@aabadie changes addressed, and commits squashed. :+1:

**Pull request merged**